common/cassandra/cassandraCluster.go

```diff
@@ -61,33 +62,50 @@ func NewCassandraCluster(cfg config.Cassandra) (*gocql.ClusterConfig, error) {
 	}
 	if cfg.TLS != nil && cfg.TLS.Enabled {
 		cluster.SslOpts = &gocql.SslOptions{
-			CertPath:               cfg.TLS.CertFile,
-			KeyPath:                cfg.TLS.KeyFile,
 			CaPath:                 cfg.TLS.CaFile,
 			EnableHostVerification: cfg.TLS.EnableHostVerification,
+			Config:                 auth.NewTLSConfigForServer(cfg.TLS.ServerName, cfg.TLS.EnableHostVerification),
+		}
+
+		var certBytes []byte
+		var keyBytes []byte
+		var err error
 
-			Config: auth.NewTLSConfigForServer(cfg.TLS.ServerName, cfg.TLS.EnableHostVerification),
+		if cfg.TLS.CertFile != "" {
+			certBytes, err = ioutil.ReadFile(cfg.TLS.CertFile)
+			if err != nil {
+				return nil, fmt.Errorf("error reading client certificate file: %w", err)
+			}
 		}
 
 		if cfg.TLS.CertData != "" {
-			certBytes, err := base64.StdEncoding.DecodeString(cfg.TLS.CertData)
+			certBytes, err = base64.StdEncoding.DecodeString(cfg.TLS.CertData)
 			if err != nil {
 				return nil, fmt.Errorf("client certificate could not be decoded: %w", err)
 			}
+		}
 
-			keyBytes, err := base64.StdEncoding.DecodeString(cfg.TLS.KeyData)
+		if cfg.TLS.KeyFile != "" {
+			keyBytes, err = ioutil.ReadFile(cfg.TLS.KeyFile)
 			if err != nil {
-				return nil, fmt.Errorf("client certificate private key could not be decoded: %w", err)
+				return nil, fmt.Errorf("error reading client certificate private key file: %w", err)
 			}
+		}
 
-			clientCert, err := tls.X509KeyPair(certBytes, keyBytes)
+		if cfg.TLS.KeyData != "" {
+			keyBytes, err = base64.StdEncoding.DecodeString(cfg.TLS.KeyData)
```

Comment:
it seems to me like silently overwriting this could lead to developers not being sure where their cert/key is coming from (whether file or data). maybe for this and cert:
```suggestion
			if keyBytes != nil {
				return nil, errors.New("cannot set both KeyFile and KeyData")
			}
			keyBytes, err = base64.StdEncoding.DecodeString(cfg.TLS.KeyData)
```

Reply:
I can make this change, although technically its breaking as current behavior today is to override. This is a super-niche case though, so if we can get consensus on this, I'll implement this suggestion

Reply:
fixed